### PR TITLE
Fix test failure of truncated time

### PR DESCRIPTION
**Pull request by dtaniwaki**
**What this PR does / why we need it**:

The test of `TestFsStoreAssignedModified` in `pkg/kubelet/kubeletconfig/checkpoint/store` fails in my environment like below.

```
$ make test WHAT=./pkg/kubelet/kubeletconfig/checkpoint/store/
Running tests for APIVersion: v1,admissionregistration.k8s.io/v1alpha1,admissionregistration.k8s.io/v1beta1,admission.k8s.io/v1beta1,apps/v1beta1,apps/v1beta2,apps/v1,authentication.k8s.io/v1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1,authorization.k8s.io/v1beta1,autoscaling/v1,autoscaling/v2beta1,batch/v1,batch/v1beta1,batch/v2alpha1,certificates.k8s.io/v1beta1,coordination.k8s.io/v1beta1,extensions/v1beta1,events.k8s.io/v1beta1,imagepolicy.k8s.io/v1alpha1,networking.k8s.io/v1,policy/v1beta1,rbac.authorization.k8s.io/v1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,scheduling.k8s.io/v1alpha1,scheduling.k8s.io/v1beta1,settings.k8s.io/v1alpha1,storage.k8s.io/v1beta1,storage.k8s.io/v1,storage.k8s.io/v1alpha1,
+++ [0628 22:53:39] Running tests without code coverage
--- FAIL: TestFsStoreAssignedModified (0.00s)
        fsstore_test.go:316: expect "2018-06-28T22:53:43+09:00" but got "2018-06-28T22:53:43+09:00"
FAIL
FAIL    k8s.io/kubernetes/pkg/kubelet/kubeletconfig/checkpoint/store    0.236s
make: *** [test] Error 1
```

My environment is
OS: macOS Sierra Version 10.12.6
File System: Journaled HFS+

The error message confused me because the comparing times looked the same in the error log. If we know certain systems truncate times, I think we can just compare less precise times to avoid confusions in tests.

**Special notes for your reviewer**:
N/A

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/kubeletconfig/checkpoint/store/fsstore_test.go [hunk 1]
```diff
@@ -306,13 +306,12 @@ func TestFsStoreAssignedModified(t *testing.T) {
...
 
-	// for now we hope that the system won't truncate the time to a less precise unit,
-	// if this test fails on certain systems that may be the reason.
+	// compare the times with RFC3339 format because certain systems truncate the time to a less precise unit.
 	modTime, err := store.AssignedModified()
 	if err != nil {
 		t.Fatalf("unable to determine modification time of assigned config source, error: %v", err)
 	}
-	if !now.Equal(modTime) {
+	if now.Format(time.RFC3339) != modTime.Format(time.RFC3339) {
```
Comment (mtaufen) [inline pkg/kubelet/kubeletconfig/checkpoint/store/fsstore_test.go]: Would it be better to just pick a time for "now" that has the least-common-precision across systems?

Comment (dtaniwaki) [reply pkg/kubelet/kubeletconfig/checkpoint/store/fsstore_test.go]: Well, indeed. I will fix it. Thanks!

**Comment by mtaufen** [comment]
Chtimes just manually sets the expected timestamp on the file backing the store so that we know what AssignedModified should return. Would be fine to do `now := time.Now().Round(time.Second)`.

**Comment by dtaniwaki** [comment]
@mtaufen I fixed it, thanks.

File: pkg/kubelet/kubeletconfig/checkpoint/store/fsstore_test.go [hunk 2]
```diff
@@ -300,14 +300,13 @@ func TestFsStoreAssignedModified(t *testing.T) {
...
-	now := time.Now()
+	now := time.Now().Round(time.Second)
 	err = store.fs.Chtimes(store.metaPath(assignedFile), now, now)
 	if err != nil {
 		t.Fatalf("could not change timestamps, error: %v", err)
 	}
 
-	// for now we hope that the system won't truncate the time to a less precise unit,
-	// if this test fails on certain systems that may be the reason.
+	// compare the times with less precise unit because certain systems truncate the time to a less precise unit.
```
Comment (mtaufen) [inline pkg/kubelet/kubeletconfig/checkpoint/store/fsstore_test.go]: Can you move this comment to where we round the time?
Since it's not next to the comparison anymore, maybe reword to `round time to the nearest second because some file systems do not support sub-second precision`.

**Comment by dtaniwaki** [comment]
@mtaufen Thank you for assisting the change many times!

**Comment by mtaufen** [comment]
Sure thing. Thanks for the fix :).

/lgtm
